pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentDispatcherMultipleConsumers.java

```diff
@@ -89,7 +89,7 @@ public class PersistentDispatcherMultipleConsumers extends AbstractDispatcherMul
     protected volatile PositionImpl minReplayedPosition = null;
     protected boolean shouldRewindBeforeReadingOrReplaying = false;
     protected final String name;
-
+    protected boolean sendInProgress;
```

Comment:
I think this flag should be `volatile`?

Reply:
Itbis guarded by synchronized blocks. No need to makenit volatile

Reply:
`synchronized` handles lock and `volatile` handles visibility, see here: 
https://stackoverflow.com/questions/3519664/difference-between-volatile-and-synchronized-in-java

I still think this flag should be `volatile`, to ensure all threads can observe the change of this flag immediately.

Reply:
I misunderstood something, you're right.